### DISCUSSION TITLE
How to run "Docker container actions"?

### DISCUSSION THREAD

**Discussion by aleskxyz** [actions/actions-runner-controller#3352] [Questions]
Hi
I have installed ARC in kubernetes mode and it is working as expected but when I want to use "Docker container actions" (for example aevea/action-kaniko) I get this error:
```
Error: Error: Building container actions is not currently supported
```
Do you know any workaround for it?

**Comment by ScottCruzen0** [comment]
action-kaniko uses a Dockerfile https://github.com/aevea/action-kaniko/blob/8de7c88b27e89ec2a3087d3351b2fee0ced7d8f8/action.yml#L65 , which GHA on VMs (non-k8s) builds on the fly prior to running the action. So one potential workaround is to fork action-kaniko and make it use an image that you've built and pushed to your OCI repo.

But I think a better solution is to not use the action, and instead use the technique described in the following doc, see the Build Workflow section:

https://dev.to/ipo/using-kaniko-to-build-and-publish-container-image-with-github-action-on-github-self-hosted-runners-d5m

**Comment by shihyuho** [comment]
Same here. When I switch to [`dind`](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/deploying-runner-scale-sets-with-actions-runner-controller#using-docker-in-docker-or-kubernetes-mode-for-containers) mode, everything goes fine.

**Comment by aleskxyz** [comment]
Thanks @ScottCruzen0 and @shihyuho for you replys
I prefer to use kubernetes mode because of security concerns.
So I could finally build my docker image with help of podman actions (was not so easy!) on ARC and I created a composite action from it.
You can take a look at it or maybe test it and give me your feedbacks!
https://github.com/marketplace/actions/build-push
Thanks!
